## 練習時間
資料的操作有很多，接下來的馬拉松中我們會介紹常被使用到的操作，參加者不妨先自行想像一下，第一次看到資料，我們一般會想知道什麼訊息？

#### Ex: 如何知道資料的 row 數以及 column 數、有什麼欄位、多少欄位、如何截取部分的資料等等

有了對資料的好奇之後，我們又怎麼通過程式碼來達成我們的目的呢？

#### 可參考該[基礎教材](https://bookdata.readthedocs.io/en/latest/base/01_pandas.html#DataFrame-%E5%85%A5%E9%97%A8)或自行 google

In [1]:
import os
import numpy as np
import pandas as pd

In [2]:
# 設定 data_path
dir_data = './data/'

In [3]:
f_app = os.path.join(dir_data, 'application_train.csv')
print('Path of read in data: %s' % (f_app))
app_train = pd.read_csv(f_app)

Path of read in data: ./data/application_train.csv


### 如果沒有想法，可以先嘗試找出剛剛例子中提到的問題的答案
#### 資料的 row 數以及 column 數

In [4]:
app_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 307511 entries, 0 to 307510
Columns: 122 entries, SK_ID_CURR to AMT_REQ_CREDIT_BUREAU_YEAR
dtypes: float64(65), int64(41), object(16)
memory usage: 286.2+ MB


#### 列出所有欄位
## 不知道是否太長? 按下RUN Cells完全跑不出來 只列122行是可以 但是所有rows(307511)就列不出來了

In [ ]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
app_train

#### 截取部分資料

In [7]:
app_train['NAME_CONTRACT_TYPE']

0              Cash loans
1              Cash loans
2         Revolving loans
3              Cash loans
4              Cash loans
5              Cash loans
6              Cash loans
7              Cash loans
8              Cash loans
9         Revolving loans
10             Cash loans
11             Cash loans
12             Cash loans
13             Cash loans
14             Cash loans
15             Cash loans
16             Cash loans
17        Revolving loans
18        Revolving loans
19             Cash loans
20        Revolving loans
21             Cash loans
22             Cash loans
23             Cash loans
24             Cash loans
25             Cash loans
26             Cash loans
27             Cash loans
28             Cash loans
29        Revolving loans
               ...       
307481         Cash loans
307482         Cash loans
307483         Cash loans
307484         Cash loans
307485    Revolving loans
307486         Cash loans
307487         Cash loans
307488      

#### 還有各種數之不盡的資料操作，重點還是取決於實務中遇到的狀況和你想問的問題，在馬拉松中我們也會陸續提到更多例子